env/types.go

```diff
@@ -0,0 +1,185 @@
+package env
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type (
+	eVar struct {
+		Key   string
+		Raw   string
+		IsSet bool
+	}
+
+	BooleanEnvVar struct {
+		eVar
+		Value bool
+	}
+
+	IntEnvVar struct {
+		eVar
+		Value int
+	}
+
+	StringEnvVar struct {
+		eVar
+		Value string
+	}
+
+	SliceEnvVar struct {
+		eVar
+		Value []string
+	}
+
+	MapEnvVar struct {
+		eVar
+		Value map[string]interface{}
```

Comment:
```suggestion
		Value map[string]string
```

Isn't it always a map of `string` to `string`?

Reply:
Yes it's true, but I changed to interface{} to match the metadata map type, to use it with `addToMapIfEmpty` func